### DISCUSSION TITLE
Premium features are a bad precedent for GitLens

### DISCUSSION THREAD

**Discussion by ghost** [gitkraken/vscode-gitlens#1860] [Feedback]
What do you guys think about them? I understand the reason behind them, but I don't like them.

**Comment by Costallat** [comment]
Don't like too, today the premium is just this and that feature, tomorrow will be all new features, next will be very basic features and all new cool stuff you will have to pay, it would be better to think another way to monetize the extension. I hate the 'freemium' model, you can get the Tibia game for example, a long time ago when you were a free player was okay, but today if you don't pay you will be nothing on the game, that's my concern about freemium models.

The price for premium features is not expensive, but I will not support this, I will keep stuck on v11.7.0 waiting for a fork/new extension with few features(but with a real open/free license), even if it's gets slow upgrades, but at least I will support a free license extension.

I would think more in charging features for companies, like the 'Team collaboration features – coming soon', things that make sense for companies to pay to help their team, and that can use some cloud service from GitKraken, but features that are local and used just by the 'local developer' won't make sense to charge for that, it's not all developers that can pay for that. I would also split into two extensions, GitLens and GitLensPremium, so I don't have to change the license and mix things on an already very great and well know extension.

I'm grateful for all the effort that Eric Amodio put on this extension, but I didn't like the new things, starting from changing the license.

**Reply by Costallat** [reply]
"You agree that GitKraken and/or
its licensors (as applicable) retain all right, title and interest in and to all
such modifications and/or patches, and all such modifications and/or patches may
only be used, copied, modified, displayed, distributed, or otherwise exploited
with a valid subscription for the correct number of user seats for the Software.
In furtherance of the foregoing, you hereby assign to GitKraken all such
modifications and/or patches. You are not granted any other rights beyond what
is expressly stated herein. Except as set forth above, it is forbidden to copy,
merge, publish, distribute, sublicense, and/or sell the Software."

**Reply by eamodio** [reply]
To be clear that license ONLY applies to code/contributions within the `plus` folders (of which there are 2 in the repository). Everything else is MIT as it has always been.

**Reply by eamodio** [reply]
Also to be 100% clear on our plans for GitLens:

GitLens is still completely free and will continue to be free forever including new features and updates, without an account or anything. Our intention and vision with GitLens and GitLens+ is to create a sustainable model to continue to heavily invest in the free and open core of GitLens – Git productivity, safety, and ease of use, and more, while also creating a set of additional, but completely optional, GitLens+ features specifically around teams, collaboration, and visualization, that only require a free account for local or public repos, and a paid account for private repos.

We will NOT be taking away features or charging for existing ones. Even in GitLens 12 the VAST majority of the work was the major refactor of the internals of GitLens to support virtual repositories (GitHub right now) and bring GitLens to the web with vscode.dev & github.dev. And as part of that refactoring, I focused a LOT on performance and efficiency from GitLens startup, git discovery, repo discovery, git parsing, blame generation, etc, etc. I also spent a lot of time improving existing UX and enhancing many existing flows in the Git Command Palette (taking advantage of many new VS Code api features). And there were the additions of autolinked issues/prs in comparisons and comparing revisions with your stashes. All of which are 100% free.

As we go forward, you will continue to see us invest heavily in the current and constantly expanding and evolving core of GitLens, as well as the optional GitLens+ features.

Hope that helps clarify things.

**Reply by Costallat** [reply]
Hi @eamodio,

Thanks for the clarification as handling/understanding with LICENSEs is something very tricky.

You answer helped to relief my concerns about the future, so I will give another try to GitLens.

Thanks,

**Reply by eamodio** [reply]
@Costallat Thank you, I really appreciate that! ❤️

**Reply by ghost** [reply]
I still am not sure about the freemium system. Thanks for responding tho

**Reply by eamodio** [reply]
@dragonDScript Would you kindly elaborate on your concerns? I would love to have a better understanding to not only incorporate with our thinking as we move forward, but also if I could help alleviate certain concerns. Thanks!

**Comment by cesaramirezfr** [comment]
Something that bugged me today was to find a new extension by GitKraken installed without my consent on my editor, what's with that?

**Reply by Costallat** [reply]
It's used by GitLens to enable the premium features, but it will be/already was removed on next release, I don't know because I freezed the version here on v11.7.0, don't want the Kraken putting their tentacles on my stuff

**Reply by eamodio** [reply]
Yes, it was removed as of the 12.0.1 release.
